activerecord/lib/active_record/connection_handling.rb

```diff
@@ -44,10 +41,9 @@ module ConnectionHandling
     #
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
-    def establish_connection(config = nil)
+    def establish_connection(config = :primary)
```

Comment:
Agreed, it's too broad. Perhaps I should just learn to stop worrying and love `self.connection_specification_name =` 😄